### DISCUSSION TITLE
Incompatibility between .Skia.GTK platform and UWP platform

### DISCUSSION THREAD

**Discussion by HowardPWeiss** [unoplatform/uno#12832] [Q&A]
The following code works in the .Skia.GTK platform but does not work in UWP platform

            var fileOpenPicker = new FileOpenPicker();
            fileOpenPicker.SuggestedStartLocation = PickerLocationId.ComputerFolder;
            fileOpenPicker.FileTypeFilter.Add(".txt");
            fileOpenPicker.FileTypeFilter.Add(".csv");
            StorageFile pickedFile = await fileOpenPicker.PickSingleFileAsync();
            if (pickedFile != null)
            {
                  string[] gameboard = File.ReadAllLines(pickedFile.Path);    

On the UWP Platfporm, this code throws the exception "Access to the path '*' is denied."

On the other hand, the following code does work on both platforms

            var fileOpenPicker = new FileOpenPicker();
            fileOpenPicker.SuggestedStartLocation = PickerLocationId.ComputerFolder;
            fileOpenPicker.FileTypeFilter.Add(".txt");
            fileOpenPicker.FileTypeFilter.Add(".csv");
            StorageFile pickedFile = await fileOpenPicker.PickSingleFileAsync();
            if (pickedFile != null)
            {
                try
                {
                    var gameboardX = await Windows.Storage.FileIO.ReadLinesAsync(pickedFile);

**Comment by FrozDark** [comment]
That is correct. UWP is more permission required by the underlying platform specifics. Permissions required to access a file on UWP and is resolved by having a handler to a file and working with specific FileIO API. While GTK is native system process and have IO access to almost everything as it is not restricted by the system
